contrib/mesos/pkg/minion/server.go

```diff
@@ -243,6 +252,22 @@ func (ms *MinionServer) Run(hks hyperkube.Interface, _ []string) error {
 		log.Fatalf("No API client: %v", err)
 	}
 	ms.clientConfig = clientConfig
+	c := client.NewOrDie(clientConfig)
+
+	// create or update node object on apiserver. This must be done synchronously
+	// because the kubelet will refuse launching pods if the labels do not match.
+	labels := map[string]string{}
+	if ms.nodeLabelString != "" {
+		err = json.Unmarshal([]byte(ms.nodeLabelString), &labels)
+		if err != nil {
+			log.Fatalf("Invalid node-labels string: %v", err)
+		}
+	}
+	log.Infof("Creating or updating node %s with labels %v", ms.nodeHostName, labels)
+	_, err = node.CreateOrUpdate(c, ms.nodeHostName, labels)
```

Comment:
Forget to say that I don't like those two choice for node creation in the executor of course ;-)

The kubelet modification might be an option. To avoid polling one could do a short-term watch instead to avoid load.